config/v1/types_network.go

```diff
@@ -13,20 +13,50 @@ type Network struct {
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// spec holds user settable values for configuration
+	// Spec holds user settable values for configuration.
 	Spec NetworkSpec `json:"spec"`
-	// status holds observed values from the cluster. They may not be overridden.
+	// Status holds observed values from the cluster. They may not be overridden.
 	Status NetworkStatus `json:"status"`
 }
 
+// NetworkSpec is the desired network configuration.
+// As a general rule, this SHOULD NOT be used directly. Instead, you should
+// consume the NetworkStatus, as it indicates the currently deployed configuration.
 type NetworkSpec struct {
-	// serviceCIDR
-	// servicePortRange
-	// vxlanPort
-	// 	ClusterNetworks    []ClusterNetworkEntry `json:"clusterNetworks"`
+	// IP address pool to use for pod IPs.
+	ClusterNetworks []ClusterNetwork `json:"clusterNetworks"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	ServiceNetworks []string `json:"serviceNetworks"`
+
+	// NetworkType is the plugin that is to be deployed (e.g. OpenShiftSDN).
+	// N.B. changing this is currently not supported.
+	NetworkType string `json:"networkType"`
 }
 
+// NetworkStatus is the current network configuration.
 type NetworkStatus struct {
+	// IP address pool to use for pod IPs.
```

Comment:
Not sure when `ObservedGeneration` is required. cc @deads2k

Reply:
I don't think we require `OG` in config.

Reply:
> Not sure when `ObservedGeneration` is required. cc @deads2k

We need it when we're reconciling the workload resources to a particular level.  I don't anticipate seeing it in these types.